contracts/interchain-token-service-impl.clar

```diff
@@ -687,6 +667,21 @@
         (try! (contract-call? .interchain-token-service-storage
             emit-interchain-token-id-claimed token-id deployer salt
         ))
+        (try! (contract-call? .verify-onchain verify-nit-deployment
+            (get nonce verification-params)
+            (get fee-rate verification-params)
+            (get signature verification-params)
+            (get contract-name contract-principal)
+            (get deployer contract-principal)
+            (get proof verification-params)
+            (get tx-block-height verification-params)
+            (get block-header-without-signer-signatures verification-params)
+        ))
+
+        (try! (contract-call? token setup token-id (some minter-unpacked) name symbol
+            decimals none (some minter-unpacked)
+        ))
+
         ;; #[filter(verification-params, minter-unpacked, supply)]
         (try! (native-interchain-token-checks token minter-unpacked token-id supply
```

Comment:
I think we no longer need the `native-interchain-token-checks` since it just checked token information that we already know now is correct.